gremlin/ui/device.py

```diff
@@ -30,13 +31,108 @@
 from gremlin import common, event_handler, joystick_handling, shared_state
 from gremlin.error import GremlinError
 from gremlin.intermediate_output import IntermediateOutput
-from gremlin.types import InputType
+from gremlin.types import InputType, PropertyType
+from gremlin.util import file_exists_and_is_accessible
+from gremlin.common import SingletonDecorator
+from gremlin.config import Configuration
 
 
 QML_IMPORT_NAME = "Gremlin.Device"
 QML_IMPORT_MAJOR_VERSION = 1
 
 
+@SingletonDecorator
+class DeviceDatabase:
+    """Provides button/axis to name mapping for known devices"""
+
+    def __init__(self) -> None:
+        self._load()
+
+    def _load(self) -> None:
+
+        db_file = "device_db.json"
+        if not file_exists_and_is_accessible(db_file):
+            return
+
+        load_successful = False
+        json_data = {}
+        with open(db_file) as hdl:
```

Comment:
Is there any benefit to this over a more concise `json_data = json.load(open(...))` in a try except block?

Reply:
No, I usually used json.load(), but saw this method in gremlin/config.py and reused it. If json.load() is preferred, I can refactor it.

Reply:
That code probably was written eons ago and I wouldn't know why I decided to do it that way, so I think doing the json.load() bit is better as that's what I'd gravitate towards nowadays too.

Reply:
OK, will refactor.